include/quda_matrix.h

```diff
@@ -1309,6 +1308,84 @@ namespace quda {
       return;
     }
 
+    /**
+       Direct port of the TIFR expsu3 algorithm
+    */
+    template <typename Float> __device__ __host__ void expsu3(Matrix<complex<Float>, 3> &q)
+    {
+      typedef complex<Float> Complex;
```

Comment:
I just assume that this has been tested somehow ...

Reply:
It's just been moved from gauge_update_quda directly.  And yes, it's been tested, since this is what is used to exponentiate the Lie algebra field when creating random SU(3) fields from Gaussian a distributed su(3) field.

Reply:
I moved the function from the gauge update routine to the header so it could be used in different files.